File: src/main/java/kr/co/bookand/backend/feedback/domain/Feedback.java
package kr.co.bookand.backend.feedback.domain;

import kr.co.bookand.backend.account.domain.Account;
import kr.co.bookand.backend.common.domain.BaseEntity;
import lombok.*;
import org.springframework.lang.Nullable;

import javax.persistence.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Feedback extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "feedback_id")
    private Long id;

    @Enumerated(EnumType.STRING)
    private FeedbackType feedbackType;

    @Enumerated(EnumType.STRING)
    @Nullable
    private FeedbackTarget feedbackTarget;

    @Column(length = 1000)
    private String content;

    @ManyToOne(fetch = FetchType.LAZY)
    private Account account;
}

File: src/main/kotlin/kr/co/bookand/backend/feedback/domain/Feedback.kt
package kr.co.bookand.backend.feedback.domain

import kr.co.bookand.backend.account.domain.Account
import kr.co.bookand.backend.common.domain.BaseEntity
import kr.co.bookand.backend.feedback.domain.dto.CreateFeedbackRequest
import org.springframework.lang.Nullable
import javax.persistence.*

@Entity
class Feedback(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "feedback_id")
    var id: Long = 0,

    @Enumerated(EnumType.STRING)
    var feedbackType: FeedbackType,

    @Enumerated(EnumType.STRING)
    @Nullable
    var feedbackTarget: FeedbackTarget,

    @Column(length = 1000)
    var content: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id")
    var account: Account? = null

) : BaseEntity() {

    constructor(createFeedbackRequest: CreateFeedbackRequest) : this(
        feedbackType = FeedbackType.valueOf(createFeedbackRequest.feedbackType),
        feedbackTarget = FeedbackTarget.valueOf(createFeedbackRequest.feedbackTarget),
        content = createFeedbackRequest.content,
    )
}